### PR TITLE
Add `types` param to `GET /api/v1/notifications` in REST API

### PR DESCRIPTION
**Pull request by Gargron**
Instead of specifying every known type to exclude, you can specify only the types you want.

### CODE REVIEW

File: spec/controllers/api/v1/notifications_controller_spec.rb [hunk 1]
```diff
@@ -95,28 +95,12 @@
...
-      end
-
-      it 'excludes reblog' do
-        expect(assigns(:notifications).map(&:activity)).to_not include(@reblog_of_first_status)
-      end
-
-      it 'excludes follow' do
-        expect(assigns(:notifications).map(&:activity)).to_not include(@follow)
+      it 'returns only notifications from specified user' do
+        expect(assigns(:notifications).map(&:from_account_id).uniq).to eq [third.account.id]
```
Comment (ClearlyClaire) [inline spec/controllers/api/v1/notifications_controller_spec.rb:99]: It's not testing that it returns any notification anymore…

File: spec/controllers/api/v1/notifications_controller_spec.rb [hunk 2]
```diff
@@ -125,54 +109,36 @@
...
       end
 
-      it 'includes reblog' do
-        expect(assigns(:notifications).map(&:activity)).to include(@reblog_of_first_status)
-      end
-
-      it 'excludes mention' do
-        expect(assigns(:notifications).map(&:activity)).to_not include(@mention_from_status)
+      it 'returns everything but excluded type' do
+        expect(assigns(:notifications).map(&:type).uniq).to_not include(:mention)
```
Comment (ClearlyClaire) [inline spec/controllers/api/v1/notifications_controller_spec.rb:127]: doesn't actually test it returns anything

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]
I think some tests got inadequately changed, but otherwise it looks good to me.

File: spec/controllers/api/v1/notifications_controller_spec.rb (hunk 1 above)
Comment (Gargron) [reply spec/controllers/api/v1/notifications_controller_spec.rb:99]: If it returned none, it would not equal the array

**Comment by ClearlyClaire** [comment]
Not on my computer right now so I cannot formally accept the changes but they look fine

**Pull request merged**